department-of-veterans-affairs/vets-api
# [API-38809] POA v2 - Appoint POA for dependent claimants

## Summary

In the v2 /2122 and /2122a submissions: when the `lighthouse_claims_api_poa_dependent_claimants` feature is enabled and `claimant` is included in the request, appoint the POA to the dependent claimant.

## Related issue(s)

[API-38809](https://jira.devops.va.gov/browse/API-38809)

## Testing done


### Test setup
We need to mock MPI since our claimant (i.e., dependent) is not available in the `vets-api-mockdata` repo, and we lookup the claimantʼs details by ICN via MPI. To prepare:
1. Checkout this PRʼs branch.
1. Add `byebug` to [`power_of_attorney_validation.rb#L36`](https://github.com/department-of-veterans-affairs/vets-api/blob/5f57559f841341182dcd5507a4ab8b884a5ef4a4/modules/claims_api/app/controllers/concerns/claims_api/v2/power_of_attorney_validation.rb#L36).
1. Add `byebug` to [`power_of_attorney/base_controller.rb#L77`](https://github.com/department-of-veterans-affairs/vets-api/blob/5f57559f841341182dcd5507a4ab8b884a5ef4a4/modules/claims_api/app/controllers/claims_api/v2/veterans/power_of_attorney/base_controller.rb#L77).

### Test steps
1. Pick a test veteran with dependents (use VBMS to see dependents).
1. Pick any dependent from Step 1.
1. Find the dependentʼs ICN. Use a staging Argo console and run the following:
   ```
   MPI::Service.new.find_profile_by_attributes(first_name: "LILLY", last_name: "GRAY", ssn: "545787121", birth_date: "05/01/2015")
   ```
   ☝️ replace test data with the dependentʼs data found in Step 2.
1. Send the following request to a local server running this PRʼs branch:
    ```
    curl --location 'http://localhost:3000/services/claims/v2/veterans/1012666073V986297/2122' \
    --header 'Content-Type: application/json' \
    --header 'Authorization: ••••••' \
    --data '{
        "data": {
            "attributes": {
                "veteran": {
                    "address": {
                        "addressLine1": "123",
                        "city": "city",
                        "stateCode": "OR",
                        "country": "US",
                        "zipCode": "12345"
                    }
                },
                "serviceOrganization": {
                    "poaCode": "083",
                    "registrationNumber": "12345"
                },
                "claimant": {
                    "claimantId": "1012880712V416406",
                    "address": {
                        "addressLine1": "123 anystreet",
                        "city": "anytown",
                        "stateCode": "OR",
                        "country": "USA",
                        "zipCode": "12345"
                    },
                    "relationship": "Child"
                }
            }
        }
    }'
    ```
    - Replace the ICN in the URL with the test veteranʼs ICN.
    - Replace the `claimantId` with the dependentʼs ICN found in Step 3.
    - Your local database must have the `poaCode` and `registrationNumber` combination or the request will fail.
1. We hit our first `byebug`. Mock the MPI profile by running:
    ```
    (byebug) Claimant = Struct.new(:given_names, :family_name, :participant_id)
    (byebug) claimant = Claimant.new(['lilly'], 'gray', '600145005')
    ```
    ☝️ replace test data with dependentʼs data found in Step 2.
1. We hit our second `byebug`. Mock the MPI profile by running:
    ```
    (byebug) Claimant = Struct.new(:participant_id, :ssn)
    (byebug) claimant = Claimant.new('600145005', '545787121')
    ```
    ☝️ replace test data with dependentʼs data found in Step 2.
1. Continue.
1. Observe the successful response.
1. Verify in VBMS that the dependentʼs POA is assigned to the requestʼs POA from Step 4.
    
## Screenshots
**Before**
![before](https://github.com/user-attachments/assets/707de624-e705-4145-abfe-312fa6750818)

**After**
![after](https://github.com/user-attachments/assets/993eaf13-521a-4fad-a125-4cd49f606ddb)

## What areas of the site does it impact?

POA v2 submission (/2122 and /2122a)

## Acceptance criteria


## Requested Feedback

We are mocking MPI in our local testing due to system limitations. Ideally, we should turn on this feature in staging and verify the assignments work as expected before enabling in production.
